phoenix-core/src/main/java/org/apache/phoenix/calcite/rules/PhoenixConverterRules.java

```diff
@@ -232,8 +232,7 @@ public RelNode convert(RelNode rel) {
         private static Predicate<LogicalSort> HAS_FETCH = new Predicate<LogicalSort>() {
             @Override
             public boolean apply(LogicalSort input) {
-                return input.offset == null 
-                        && input.fetch != null;
+                return input.fetch != null;
```

Comment:
With respect to my comment below, if there is a limit, then we apply the rule, otherwise we do not (regardless of offset).
 I may have the wrong idea here, but based on my understanding of offset, all of the test cases pass as expected. Are the test cases I have laid out in CalciteIT what you would expect?